### PR TITLE
Feature/image service grails3

### PR DESCRIPTION
**Pull request by djtfmartin**

### CODE REVIEW

File: ansible/roles/image-service/templates/sql/image-service-export.sql [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            split_part(original_filename, '||', 2) AS "occurrenceID",
+            CONCAT( '{{image_service_base_url}}{{image_service_context_path}}/image/proxyImageThumbnailLarge?imageId=', image_identifier) AS identifier,
+            regexp_replace(creator, E'[\\n\\r]+', ' ', 'g' ) AS creator,
+            date_taken AS created,
+            regexp_replace(title, E'[\\n\\r]+', ' ', 'g' ) AS title,
+            mime_type AS format,
+            regexp_replace(license, E'[\\n\\r]+', ' ', 'g' ) AS license,
+            regexp_replace(rights, E'[\\n\\r]+', ' ', 'g' ) AS rights,
+            regexp_replace(rights_holder, E'[\\n\\r]+', ' ', 'g' ) AS "rightsHolder",
+            CONCAT('{{image_service_base_url}}{{image_service_context_path}}/image/', image_identifier) AS "references",
```
Comment (vjrj) [inline ansible/roles/image-service/templates/sql/image-service-export.sql]: This is somehow problematic when not using ansible (docker, debs, etc). Just for the record.

File: ansible/roles/image-service/tasks/main.yml [hunk 2]
```diff
@@ -29,87 +47,141 @@
...
-    context_path: "{{ images_context_path }}"
+    hostname: "{{ image_service_hostname }}"
+    context_path: "{{ image_service_context_path }}"
     nginx_paths:
-      - path: "{{ images_context_path }}"
+      - path: "{{ image_service_context_path }}"
         sort_label: "1_ws"
         is_proxy: true
-        proxy_pass: "http://127.0.0.1:8080{{ images_context_path }}"
+        proxy_pass: "http://127.0.0.1:8080{{ image_service_context_path }}"
```
Comment (vjrj) [inline ansible/roles/image-service/tasks/main.yml]: This generates a duplicate location /:
```
/etc/nginx/vhost_fragments/images.l-a.site/http_70_location_1_ws_70_start:    location / {
/etc/nginx/vhost_fragments/images.l-a.site/http_70_location_2_store_70_start:    location /store {
/etc/nginx/vhost_fragments/images.l-a.site/http_70_location__70_start:    location / {
```
and the nginx task fails.

File: ansible/roles/image-service/tasks/main.yml [hunk 3]
```diff
@@ -29,87 +47,141 @@
...
-    hostname: "{{ images_hostname }}"
-    context_path: "{{ images_context_path }}"
+    hostname: "{{ image_service_hostname }}"
+    context_path: "{{ image_service_context_path }}"
     nginx_paths:
-      - path: "{{ images_context_path }}"
+      - path: "{{ image_service_context_path }}"
         sort_label: "1_ws"
         is_proxy: true
-        proxy_pass: "http://127.0.0.1:8080{{ images_context_path }}"
```
Comment (vjrj) [inline ansible/roles/image-service/tasks/main.yml]: Using `8080` conflicts with servers with tomcat installed (like demo playbook). 

I think you should use other different port like CAS services do:
https://github.com/AtlasOfLivingAustralia/ala-install/blob/1bb42eb49caa3a8246686fb1149856e80e81fc05/ansible/aws-cas-5.yml#L181

**Comment by vjrj** [comment]
About the nginx duplicated location issue: It's in part an issue with `nginx_vhost` role (and ansible 2.8 
https://github.com/AtlasOfLivingAustralia/ala-install/issues/298).

Attached [some patch to fix it](https://github.com/AtlasOfLivingAustralia/ala-install/files/3656601/image-service-port.patch.txt).

Also some [logs with some Hibernate errors I get](https://github.com/AtlasOfLivingAustralia/ala-install/files/3656602/image-service.log).

**Comment by ansell** [comment]
@vjrj The nginx_vhost patch doesn't look as expected, but I will work through it. It is likely another instance of nginx_vhost that isn't customising `nginx_paths` that needs to be modified, but I am not familiar with where it is right now.

**Comment by ansell** [comment]
Just to clarify, when you are testing, are you using `nginx_vhost_fragments_to_clear` in your inventory?

**Comment by ansell** [comment]
@vjrj There were two calls to `nginx_vhost` from within `image-service/tasks/main.yml`. I deleted the second one so the `nginx_paths` are managed in a single call to `nginx_vhost`. https://github.com/AtlasOfLivingAustralia/ala-install/pull/333/commits/bc9c7b4ef89ed5f7cb3ef765ffc6abd61a85aaf0

Future debugging of `nginx_paths`/`nginx_vhost` multiple calls can be done by adding ` --extra-vars "nginx_paths_debug_enabled=True"` to the end of your `ansible-playbook` command. That will make it pause during `nginx_vhost` and display the paths it is about to use, so you can identify if there are unexpected calls into it.

**Comment by vjrj** [comment]
@ansell, you are right about the dup location (I have pending to send some PR for some `is defined` [bool variable that is not checked as bool](https://github.com/AtlasOfLivingAustralia/ala-install/blob/94c733275e7857aa96b3e17e4475bb7519218a13/ansible/roles/bie-index/tasks/main.yml#L20) and I was thinking that this is other similar issue).

Also I had:
```
nginx_vhost_fragments_to_clear=["l-a.site"]
```
instead of 
```
nginx_vhost_fragments_to_clear=["images.l-a.site"]
```
because is a multi host/services inventory.

so my updated [patch](https://github.com/AtlasOfLivingAustralia/ala-install/files/3662199/image-service-port.patch2.txt) only sets the embedded tomcat port.

Other minor issue that I see @djtfmartin is that image-service is not reloaded/restarted after rerunning `ansible-playbook`.

**Comment by ansell** [comment]
@vjrj `nginx_vhost_fragments_to_clear` is an array property, so you can put as many hosts in it as necessary.

For example:

```
nginx_vhost_fragments_to_clear=["l-a.site", "images.l-a.site"]
```

I will work on the handlers so that `image-service` gets restarted when its configuration/software is updated.

**Comment by vjrj** [comment]
Thanks for the clarification. Just added to the [inventories generator](https://github.com/vjrj/generator-living-atlas/commit/eb8b4f88cb185d71079503bfa461d61c3c74e293) and I've just updated our production inventories: 
```
-nginx_vhost_fragments_to_clear=["datos.gbif.es"]
+nginx_vhost_fragments_to_clear = ["datos.gbif.es","colecciones.gbif.es","registros.gbif.es","registros-ws.gbif.es","especies.gbif.es","especies-ws.gbif.es","imagenes.gbif.es","listas.gbif.es","regiones.gbif.es","logger.gbif.es","index.gbif.es","auth.gbif.es","jenkins.gbif.es","espacial.gbif.es"]
```

**Pull request merged**